docs/user/preparing_a_pr.md

```diff
@@ -0,0 +1,80 @@
+
+# Preparing a Pull Request
+
+After adding a new feature or implementing a bug fix, you will probably want to
+merge your fix into the garage master branch and make it available to others.
+This page will guide you through the process of creating a pull request and
+merging your changes.
+
+## Precursors
+
+Assuming you read and followed the [git workflow](git_workflow.md) guide,
+you're almost ready to submit a PR. You should make sure your feature branch is
+rebased on to the latest version of `rlworkgroup/master` and that your
+pre-commit hooks pass, then run the test suite as shown below.
+
+### Passing the Test Suite
+
+Before submitting a PR, you should make sure that your changes do not cause any
+breakages in the garage codebase. We do this by running the test suite, which
+includes all the unit and integration tests that exist on the master branch,
+plus any new ones you added. You can run the test suite locally from the garage
+root directory by running
+
+```sh
+make test
+```
+
+This will run the test suite in a docker container, so be sure to have docker
+installed.
+
+## Submitting a PR
+
+Once you've completed the above steps, you are ready to submit a PR. First push
+your updated feature branch to your github fork:
+
+```sh
+git push origin your-feature-branch
+```
+
+Then, navigate to your github fork, switch to your feature branch, and click on
+`pull-request`.  On the next page, select `rlworkgroup/garage` as the base
+repository and your feature branch as the head repository. This will submit a
+PR to merge your changes into the base repository. If the changes shown look
+good, you can submit your PR.
+
+## Managing a PR
+
```

Comment:
A small suggestion is to have a section on the PR tags - one usage of tags is to categorize the PR types i.e. 'documentation', 'bugs' etc, another usage is to perform merge action automated by mergify i.e. 'ready-to-merge'.

Reply:
Added.